File: templates/matlab/materials_microstructure_surface3d.m
function fig = materials_microstructure_surface3d()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    fig = generated_template_figure('surface3d', 1818, 'materials microstructure: 3D response surface', 'materials microstructure', '3D response surface');
end

File: templates/python/materials_microstructure_surface3d.py
"""materials_microstructure_surface3d: 材料微结构三维响应曲面（3d-surface 模式，合成数据）."""
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
from generated_patterns import make_template_figure


def make_figure(title='materials microstructure: 3D response surface'):
    return make_template_figure('surface3d', seed=1818, title=title, domain='materials microstructure', topic='3D response surface')


if __name__ == '__main__':
    fig = make_figure()
    fig.savefig(__file__.replace('.py', '.png'), dpi=150)
